### DISCUSSION TITLE
Пакування в Python

### DISCUSSION THREAD

**Discussion by malor** [shopokodu/community#42] [Теми для подкасту]
1. PyPA
1. Пакування бібліотек і застосунків:
   - який інструментарій обрати?!
   - lock-файли
3. Пакування проєктів із відкритим кодом:
   - особливості курованих репозиторіїв Linux-дистрибутивів
   - sdist проти wheel
4. virtualenv
   - venv vs virtualenv
   - Навіщо virtualenv в контейнерах.
5. Хороші практики:
   - безпарольна публікація на PyPI через GitHub Actions
   - [cibuildwheel](https://github.com/pypa/cibuildwheel)
   - [pip-tools](https://github.com/jazzband/pip-tools)
   - https://discuss.python.org/c/packaging
   - setup.py
6. [PEP 668](https://peps.python.org/pep-0668/).
7. Тестування.

Матеріали:
- https://alpopkes.com/posts/python/packaging_tools/
- https://hackmd.io/@CAM-Gerlach/py-packaging-summit-2023

**Comment by webknjaz** [comment]
Згадані сьогодні штуки:
https://peps.python.org/pep-0735
https://github.com/marketplace/actions/pypi-publish
https://diataxis.fr
https://pdm-project.org/latest/usage/pep582/
https://packaging.python.org
https://github.com/jazzband/pip-tools/issues/1326#issuecomment-1834517252

Шкода, що забули пригадати про можливості контрибютити, як-от переклади: https://packaging.python.org/en/latest/contribute/#translations / https://hosted.weblate.org/projects/pypa/warehouse/.
Чи pip-tools допомагати мейнтейнити: https://jazzband.co.
